### ISSUE TITLE
CONCATPDF - PDF generation error after payment recording

### ISSUE THREAD

**Issue by atm-maxime** [DoliCloud/DoliMods#7]
**Describe the bug**
TCPDF_PARSER ERROR: Empty PDF data.

**To Reproduce**
1. Generate an invoice PDF with a PDF doc appended thanks to the module
2. Delete the PDF used in the concat PDF module setup
3. Back on the invoice, try to record a payment

**Expected behavior**
Payment should be recorded correctly

**Additional context**
I know that on payment recording, Dolibarr regenerates the PDF and with concatPDF, it tries to use the file used on the first generation. But if this file does not exist anymore, it generates the error TCPDF_PARSER ERROR: Empty PDF data. I think that if the file the system wants to use to concat with doesn't exist, it should ignore the concat part.

**Comment by eldy** [comment]
It seems problem happen only if we open the page of invoice and we don't refresh it after deleting the PDF to concat.
So the combo box with the PDF to concat is still in the form and when we send the generate, we got the error.
If after deletion of the PDF to contact, we got back on the invoice into another session or we refresh the page, the combo with list of PDF to concat does not appear and there is no error.

So case to get the error is very rare, and when we got it, retrying is enough to have error disappeared.
